samanthas_telegram_bot/api_queries/send_get.py

```diff
@@ -0,0 +1,293 @@
+
+logger = logging.getLogger(__name__)
+
+STATUS_AT_CREATION_STUDENT_TEACHER = "awaiting_offer"
+STATUS_AT_CREATION_TEACHER_UNDER_18 = "active"
+
+
+# TODO how to best test these functions?  We can't test whether all data was collected by the bot,
+#  but it may make sense to test whether all data is passed to backend (to avoid cases when we
+#  add a new field to model and forget to reflect it here).  Is there some sort of "dry run" mode
+#  in the backend so we can send a post request without the record being created?  Ofc, we can
+#  just create a record and then delete it.
+async def _send_personal_info_get_id(user_data: UserData, bot: Bot) -> int:
+    """Creates a personal info item. This function is meant to be called from other functions."""
+    async with httpx.AsyncClient() as client:
+        r = await client.post(
+            f"{API_URL_PREFIX}/personal_info/",
+            data={
+                "communication_language_mode": user_data.communication_language_in_class,
+                "first_name": user_data.first_name,
+                "last_name": user_data.last_name,
+                "telegram_username": user_data.tg_username if user_data.tg_username else "",
+                "email": user_data.email,
+                "phone": user_data.phone_number,
+                "utc_timedelta": (
+                    f"{user_data.utc_offset_hour:02d}:{user_data.utc_offset_minute}:00"
+                ),
+                "information_source": user_data.source,
+                "registration_telegram_bot_chat_id": user_data.chat_id,
+                "registration_telegram_bot_language": user_data.locale,
+            },
+        )
+    if r.status_code == httpx.codes.CREATED:
+        data = json.loads(r.content)
+        logger.info(
+            f"Chat {user_data.chat_id}: Created personal data record for {user_data.first_name} "
+            f"{user_data.last_name} ({user_data.email}), ID {data['id']}"
+        )
+        return data["id"]
+
+    await log_and_report(
+        text=(
+            f"Chat {user_data.chat_id}: Failed to create personal data record "
+            f"(code {r.status_code}, {r.content})"
+        ),
+        bot=bot,
+        parse_mode=None,
+        logger=logger,
+        level="error",
+    )
+    return 0
+
+
+async def send_student_info(update: Update, user_data: UserData) -> bool:
+    """Sends a POST request to create a student and send results of assessment if any."""
+
+    personal_info_id = await _send_personal_info_get_id(user_data=user_data, bot=update.get_bot())
+
+    data = {
+        "personal_info": personal_info_id,
+        "comment": user_data.comment,
+        "status": STATUS_AT_CREATION_STUDENT_TEACHER,
+        "status_since": _format_status_since(update),
+        "can_read_in_english": user_data.student_can_read_in_english,
+        "is_member_of_speaking_club": False,  # TODO can backend set to False by default?
```

Comment:
Of course can :)

Reply:
We'll leave it as a TODO for now, I guess